plugins/modules/host_hide.py

```diff
@@ -183,47 +243,13 @@ def main():
 
     action_name = "hide_host" if hidden else "unhide_host"
 
-    query_result = falcon.perform_action(action_name=action_name, ids=hosts)
-
-    # The API returns both successful and failed hosts in the same response. This
-    # means we need to handle errors differently than we normally would.
-    good = query_result["body"]["resources"]
-    bad = query_result["body"]["errors"]
-
-    # If we get nothing back, handle the error
-    if not good and not bad:
-        handle_return_errors(module, falcon, query_result)
-
-    # Create a mapping for passed-in host IDs to manage their states
-    host_mapping = {host_id: "" for host_id in hosts}
-
-    # For hosts in 'good', add the ID to the hosts list
-    if good:
-        result["changed"] = True
-        for host in good:
-            host_mapping[host["id"]] = host["id"]
-
-    # For hosts in 'bad', manage state and failed_hosts
-    for host in bad:
-        message, code = host["message"], host["code"]
-
-        for host_id in host_mapping.keys():
-            if host_id not in message:
-                continue
-
-            if code == 409:  # Host already in desired state
-                host_mapping[host_id] = host_id
-            else:
-                result["failed_hosts"].append(
-                    {
-                        "id": host_id,
-                        "code": code,
-                        "message": message,
-                    }
-                )
-
-    # Add the hosts to the result
-    result["hosts"] = [value for value in host_mapping.values() if value]
+    # API can only process 100 hosts at a time. Lets check if we have more than 100
+    # hosts to process and if so, we will process them in batches of 100.
+    if len(hosts) > 100:
+        for i in range(0, len(hosts), 100):
+            process_hosts(module, falcon, action_name, hosts[i : i + 100], result)
+    else:
+        process_hosts(module, falcon, action_name, hosts, result)
```

Comment:
Do you need the `for` inside an `if`? If len == 150 that would be handled the same in the second iteration of the loop as len == 50.

Reply:
No I dont.. going to fix it now along with some of the lint issues.